retrieval/scrape.go

```diff
@@ -121,8 +123,8 @@ func init() {
 // scrapePool manages scrapes for sets of targets.
 type scrapePool struct {
 	appendable Appendable
-
-	ctx context.Context
+	logger     log.Logger
+	ctx        context.Context
```

Comment:
Go nit: I see this hasn't changed from the previous commit to this one, but you're generally not supposed to store a context as a member in a struct.

Ideally, the context would instead be passed to each method of the scrapePool type on its own.

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.
https://golang.org/pkg/context/

Reply:
Yes, it's indeed not ideally but currently required due to some semantic constraints. Or at least code would get unnecessarily complex otherwise.

I meant to get rid of nits such as this one as part of #3114 but it's a huge change and would get to contentious at this point.
So for now we should just live with it I think.

Reply:
Works for me.